### PR TITLE
Admin commands

### LINKED ISSUE #39: Count people in different roles
Command that will count how many people have certain roles.

syntax
[rule] = <and|or>[not]<Role>
`.countRoles [not]<Role> [rule_1][rule_2].......

Is it possible or it will do too many requests to discord or something<

### LINKED ISSUE #66: New accounts admin commands
All commands should use only Mods.

- [x] Account erasing 

When some user creates another Discord user account while they were authenticated on the previous (lost) account, they cannot authenticate with this new account. This is good and intended.

My propose is to create such functionality that we can erase the user by its username (internally it will compute the hash and erase the record with the previous one) and the user with the new account can complete the auth process then. Something like `/user erase [CTU username]`.

- [x] Account info

We should be able to get user info in Discord, aka get the whole Database record, to see user id, authId (ctu username hash), verify status etc. Something like `user info [CTU username]`

### LINKED ISSUE #69: Basic admin commands
- [x] Send message
- [x] Edit message
- ~Send image~
- ~Edit image~
- ~Send embed~ -> #76 
- ~Edit embed~ -> #76 
- [x] React to message

### LINKED ISSUE #70: Channel commands
- [x] Clone channel
- [ ] 

All commands are for admins only.

### PR DESCRIPTION
**Pull request by tenhobi**
Adds:

- channel
  - clone
- message
  - send
  - edit
  - react
- member
  - info
  - erase
  - count

### CODE REVIEW

File: src/HonzaBotner.Discord.Services/Commands/ChannelCommands.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            string name,
+            [Description("Additional roles to view this channel")]
+            params DiscordRole[] roles)
+        {
+            DiscordChannel? cloned = null;
+            try
+            {
+                cloned = await channel.CloneAsync();
+            }
+            catch
```
Comment (ostorc) [inline src/HonzaBotner.Discord.Services/Commands/ChannelCommands.cs]: Log exception

File: src/HonzaBotner.Discord.Services/Commands/MemberCommands.cs [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+                return;
+            }
+
+            try
+            {
+                _dbContext.Verifications.Remove(databaseRecord);
+                await _dbContext.SaveChangesAsync();
+                await ctx.Channel.SendMessageAsync("Member has been erased.");
+            }
+            catch
```
Comment (ostorc) [inline src/HonzaBotner.Discord.Services/Commands/MemberCommands.cs]: Log exception

File: src/HonzaBotner.Discord.Services/Commands/MessageCommands.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                return;
+            }
+
+            foreach (DiscordEmoji emoji in emojis)
+            {
+                try
+                {
+                    await oldMessage.CreateReactionAsync(emoji);
+                }
+                catch
```
Comment (ostorc) [inline src/HonzaBotner.Discord.Services/Commands/MessageCommands.cs]: Log exception

**Review by ostorc — CHANGES_REQUESTED** [review]

File: src/HonzaBotner.Discord.Services/Commands/VoiceCommands.cs [hunk 4]
```diff
@@ -11,7 +10,7 @@
 namespace HonzaBotner.Discord.Services.Commands
 {
     [Group("voice")]
-    [Description("Commands to control custom voice channels.")]
+    [Description("Commands to control (and only works in) custom voice channels.")]
```
Comment (ostorc) [inline src/HonzaBotner.Discord.Services/Commands/VoiceCommands.cs:13]: What does custom voice channel means?

**Review by ostorc — APPROVED** [review]

File: src/HonzaBotner.Discord.Services/Commands/VoiceCommands.cs (hunk 4 above)
Comment (tenhobi) [reply src/HonzaBotner.Discord.Services/Commands/VoiceCommands.cs:13]: Those are the channels that members can create by themselves. Temporary channels. See previous issues or config.

image

**Pull request merged**